src/views/AddItem.jsx

```diff
@@ -1,3 +1,93 @@
-export function AddItem() {
-	return <p>Hello from the <code>/add-item</code> page!</p>
+import { addItem } from '../api/firebase';
+import { useState } from 'react';
+
+export function AddItem({ listId }) {
+	// declare itemName, daysUntilNextPurchase, and radioSelect
+	// into separate state variables
+	const [itemName, setItemName] = useState('');
+
+	const [daysUntilNextPurchase, setDaysUntilNextPurchase] = useState(null);
+
+	const [radioSelect, setRadioSelect] = useState('soon');
+
+	// Store values inside itemData and sent over to database
+	// otherwise log an error if request fails
+	const onFormSubmit = (event) => {
+		event.preventDefault();
+		const itemData = {
+			itemName,
+			daysUntilNextPurchase,
+		};
+		try {
+			addItem(listId, itemData);
+			alert(`${itemData.itemName} was added to your list`);
```

Comment:
I think this probably meets the acceptance criteria of telling the user that the item was added. However, the acceptance criteria of telling the user when the item was *not* successfully added is not covered here. 

Future ways to improve this could be creating and using some sort of status message component that briefly shows a message indicating whether the item was added successfully. I don't often see the use of `alert()` in many applications, though I do see the use of the related `confirm()` method to verify a user is sure they want to take a "destructive" action.

Another thing to think about in the future would be what should happen when the `addItem` call is unsuccessful. One way to simulate this is to temporarily turn off the WiFi on your laptop after you've begun running the application locally and have this form pulled up, and click the button to see how the app behaves.

A further consideration for the future would be what sorts of state change might be appropriate to the form whenever the `addItem` finishes. Should the form be reset to blank/default choices?

As the issue mentions, this form can turn into plenty of scope!

Reply:
@mtfoley Thanks, Matt! This was definitely something we brought up during office hours. I believe @djtaylor8 mentioned it was okay to just log the error (for now at least?). Lots of amazing points here. The status message component is a great idea. We were considering this, but at the time, we weren't sure if it was okay to create a completely new component that wasn't part of the criteria. But now we know it's recommended to do so! I'm itching to fix this feature now 😆

Reply:
Got it, I was around for those office hours, and now I remember that conversation!

Reply:
Great catch on error handling as part of the acceptance criteria! Yet another suggestion to handle notifications - this is a good candidate to incorporate a library (something like `react-hot-toast`). Ultimately it's up to you and your fellow devs though!

Reply:
@djtaylor8 Love the name 'hot toast'! I wanna look more into it and perhaps bring it up to collabies.

Reply:
🙌

Reply:
@mtfoley @djtaylor8 Thanks both for very thoughtful commentary.  We did see on a previous version that other teams used a new view to confirm the addition.  Something we thought we might consider changing when it comes to styling later on in development.  Also, I agree the alert seems awkward.  Will definitely look into  `confirm()`.  I did have questions about the differences between try/catch and promises as I see in the documentation for `react-hot-toast`.  Maybe I can get clarification tomorrow if my reading up today isn't clear.  Appreciate all the insight!